gene_gene/gene_interacts_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,520 @@
+
+# coding: utf-8
+
+# # Dataset Statistics for Gene Gene Sentences
+
+# This notebook is designed to show statistics on the data extracted from pubmed. The following cells below here are needed to set up the environment.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+from collections import Counter
+from itertools import product
+import os
+import pickle
+import sys
+
+sys.path.append(os.path.abspath('../../../modules'))
+
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib_venn import venn2
+import seaborn as sns
+from tqdm import tqdm_notebook
+
+sns.set(rc={'figure.figsize':(12,6), "font.size":17})
+
+
+# In[2]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[3]:
+
+
+from snorkel.models import candidate_subclass, Candidate
+GeneGene = candidate_subclass('GeneGene', ['Gene1', 'Gene2'])
+
+
+# In[4]:
+
+
+from utils.notebook_utils.dataframe_helper import write_candidates_to_excel, make_sentence_df
+
+
+# # Read Full Sentence Table
+
+# The cells below will read every sentence that contains a gene and disease entity from the sentence table in our postgres database. For time sake majority of the data has already been processed and save as files mentioned below.
+
+# In[ ]:
+
+
+sql= '''
+select id as sentence_id, text, (
+    char_length(regexp_replace(CAST(words AS TEXT), '[\u0080-\u00ff]', '', 'g')) - 
+    char_length(regexp_replace(regexp_replace(CAST(words AS TEXT), '[\u0080-\u00ff]', '', 'g'), ',', '','g'))
+) as sen_length, entity_types 
+from sentence 
+where entity_types::text like '%%Gene%%' or entity_types::text like '%%Gene%%';
+'''
+sentence_df = pd.read_sql(sql, database_str)
+sentence_df.head(2)
+
+
+# In[ ]:
+
+
+entity_data = []
+tagging_error_ids = set({})
+
+#skip tagging error
+skip_tag_error = False
+for index, row in tqdm_notebook(sentence_df.iterrows()):
+    
+    #create dictionay for mapping entity types
+    entity_mapper = {"sentence_id": row['sentence_id']}
+    
+    #Keep track of previous entity
+    previous_entity = 'o'
+    
+    #For all entitys in a given sentence decide what is tagged
+    for entity in row['entity_types']:
+        entity = entity.lower()
+
+        #Non-O tag
+        if entity != 'o' and previous_entity =='o':
+            #If entity not seen before instanciate it
+            if entity not in entity_mapper:
+                entity_mapper[entity] =0
+            entity_mapper[entity] += 1
+
+        # If previous tag was non-O and the current tag does not equal previous
```

Comment:
Why is this considered an error?

Reply:
Some sentences can contain the following tagged sequence: `O O Disease Gene Disease O O`. This is a problem, because in IOB tagging you want to have one concept per tagging block. i.e. I want disease mentions to only be next to another disease mention or an O to signify end of tagging block. If the above example were to be `O O Disease O Disease O` then it would mean there are two disease mentions and not be considered an error.